crates/types/src/data.rs

```diff
@@ -457,7 +457,8 @@ impl<TYPES: NodeType> Leaf<TYPES> {
     /// interpreted as bytes).
     #[must_use]
     pub fn genesis(instance_state: &TYPES::InstanceState) -> Self {
-        let (payload, metadata) = TYPES::BlockPayload::genesis();
```

Comment:
I think in light of this we can delete the `genesis` function entirely from the `BlockPayload` trait

Reply:
Yes but... the signatures don't currently line up. `from_transactions` takes `Arc<dyn InstanceState>` while here we have `&InstanceState`. So for the moment my solution is to do the workaround in the `genesis()` method on sequencer side. Then we should line the types up and remove `genesis` from `BlockPayload`. 

On that note, there was some discussion about this elsewhere, that maybe we shouldn't be  passing the entire state object to `from_transactions` but just the `max_block_size`. I'm not sure what future needs will be so I'm open to suggestions.